src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CaseRetriever.java

```diff
@@ -17,19 +26,43 @@
     }
 
     public CaseDetails retrieve(String jurisdiction, String caseRef) {
-        CcdAuthenticator info = factory.createForJurisdiction(jurisdiction);
-        return retrieveCase(jurisdiction, caseRef, info);
-    }
+        // not including in try catch to fast fail the method
+        CcdAuthenticator authenticator = factory.createForJurisdiction(jurisdiction);
 
-    private CaseDetails retrieveCase(String jurisdiction, String caseRef, CcdAuthenticator authenticator) {
-        return coreCaseDataApi.readForCaseWorker(
-            authenticator.getUserToken(),
-            authenticator.getServiceToken(),
-            authenticator.userDetails.getId(),
-            jurisdiction,
-            CASE_TYPE_ID,
-            caseRef
-        );
+        try {
+            CaseDetails caseDetails = coreCaseDataApi.readForCaseWorker(
+                authenticator.getUserToken(),
+                authenticator.getServiceToken(),
+                authenticator.userDetails.getId(),
+                jurisdiction,
+                CASE_TYPE_ID,
+                caseRef
+            );
+
+            logCaseDetails(caseDetails);
+
+            return caseDetails;
+        } catch (FeignException exception) {
+            if (exception.status() == NOT_FOUND.value()) {
+                log.info("Case not found. Ref:{}, jurisdiction:{}", caseRef, jurisdiction);
+
+                return null;
```

Comment:
Would optional make more sense than null here?

Reply:
just this here?

Reply:
A good choice to prevent `null` values but not leaning towards it. Unnecessary wrapper - both branches in same method call will have to be wrapped. Then envelope processor has a pre-check before hitting case retriever. So overall there are 3 branches as of this moment. It is easier to handle `null`s at this point.

For future, in case there will be no need to have information about `CaseDetail` in order to make event calls, I can see the benefit of `Optional` being used